src/middlewared/middlewared/plugins/enclosure_/enclosure_class.py

```diff
@@ -205,23 +205,32 @@ def _get_model_and_controller(self):
                 self.model = ''
                 self.controller = False
 
-    def _ignore_element(self, parsed_element_status, element):
+    def _ignore_element(self, parsed_element_status, element, element_type):
         """We ignore certain elements reported by the enclosure, for example,
         elements that report as unsupported. Our alert system polls enclosures
         for elements that report "bad" statuses and these elements need to be
         ignored. NOTE: every hardware platform is different for knowing which
         elements are to be ignored"""
         desc = element['descriptor'].lower()
-        return any((
-            (parsed_element_status.lower() == ElementStatusesToIgnore.UNSUPPORTED.value),
-            (self.is_xseries and desc == ElementDescriptorsToIgnore.ADISE0.value),
-            (self.model == JbodModels.ES60.value and desc == ElementDescriptorsToIgnore.ADS.value),
-            (not self.is_hseries and desc in (
-                ElementDescriptorsToIgnore.EMPTY.value,
-                ElementDescriptorsToIgnore.AD.value,
-                ElementDescriptorsToIgnore.DS.value,
-            )),
-        ))
+        if parsed_element_status.lower() == ElementStatusesToIgnore.UNSUPPORTED.value:
+            return True
+        elif self.is_xseries:
+            if element_type == ElementType.TEMPERATURE_SENSORS.value:
+                if self.model == ControllerModels.X10.value and desc.startswith(('dimm2', 'dimm3')):
+                    # x10 platform only has 2 DIMM slots (dimm0/1) populated so the others report
+                    # -1C temperature but the descriptor status reports as "OK".
+                    return True
```

Comment:
The x10 can _never_ and will _never_ support the dimm2 and dimm3 slots?
It would be nice if there was a better way to handle these platform specific things.

Reply:
That's what I alluded to in my comment. The dimm module isn't populated but the descriptor still reports as "OK". So there isn't a concrete way of knowing that the module isn't populated. We basically have to make some form of assumption here.